pkg/streamingpromql/planning/core/core.proto

```diff
@@ -20,6 +20,10 @@ message PositionRange {
   int64 end = 2 [(gogoproto.casttype) = "github.com/prometheus/prometheus/promql/parser/posrange.Pos"];
 }
 
+message Hints {
```

Comment:
Should this be called something like `BinaryExpressionHints`? I imagine we'll have other kinds of hints for other kinds of nodes in the future.